### DISCUSSION TITLE
New 1.29.2 is really slow

### DISCUSSION THREAD

**Discussion by djpalmis** [teslamate-org/teslamate#3975] [General]
It takes seconds to surf on new version. If I click on settings, for example, it takes about SEVENTEEN seconds to show settings page or to go back.
Is it something related to my Google cloud installation?

**Comment by cwanja** [comment]
Zero issues running on a basic 3 b+ Pi in my local network.

**Reply by djpalmis** [reply]
Thanks to your message I tried a reboot: I fixed it! Thank you

**Reply by djpalmis** [reply]
Well..  Actually it behaves a little bit weird. 
Ok, after the virtual machine reset it is faster, but, still, it takes time, more than previous version, to refresh data or loading graphana.
I don't know if it just an impression..

**Reply by cwanja** [reply]
One way to test this, since it seems like you are running a VPS, would be to either downgrade to 1.29.1 or spin up a secondary VPS with 1.29.1 and test. Restore your database as of today and see. 

I am unaware of any underlying code changes that would make it "slower" from a GUI / UX perspective.

**Reply by djpalmis** [reply]
![Screenshot_20240620_092442_Samsung Internet](https://github.com/teslamate-org/teslamate/assets/402122/12d70e4f-9a1f-4512-ae88-143d652d000c)
Honestly I'm not so skilled to manage the issue. But at the moment I can't access my page and CPU use is higher as you can see in screenshot.

**Comment by djpalmis** [comment]
Yesterday it happened again out of the blue.
I had a look at the logs.
Is it something related to a corrupted database?
Now cpu load is 50%.
Shall I reset my virtual machine? Ok but it will come back next time.. in a random way. 

![Screenshot_20240807_082453_Samsung Internet](https://github.com/user-attachments/assets/1093fca2-d255-4792-b3d6-cc262ab5ed17)

**Reply by djpalmis** [reply]
It is completely messed up. It repeats the same errors. Any support on this?:

> teslamate-1  | 2024-08-07 11:30:11.233 [info] Starting logger for 'Giaguarina'
teslamate-1  | 2024-08-07 11:30:13.006 [info] MQTT connection has been established
teslamate-1  | 2024-08-07 11:30:19.042 [info] Tzdata has updated the release from 2021e to 2024a
teslamate-1  | 2024-08-07 11:31:11.261 [error] Postgrex.Protocol (#PID<0.2444.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.2701.0> timed out because it queued and checked out the connection for longer than 60000ms
teslamate-1  | 
teslamate-1  | #PID<0.2701.0> was at location:
teslamate-1  | 
teslamate-1  |     :prim_inet.recv0/3
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:3188: Postgrex.Protocol.msg_recv/4
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:2213: Postgrex.Protocol.recv_bind/3
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:2105: Postgrex.Protocol.bind_execute/4
teslamate-1  |     (db_connection 2.6.0) lib/db_connection/holder.ex:354: DBConnection.Holder.holder_apply/4
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:1512: DBConnection.run_execute/5
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:1607: DBConnection.run/6
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:743: DBConnection.parsed_prepare_execute/5
teslamate-1  | 
teslamate-1  | 2024-08-07 11:31:12.244 [error] GenStateMachine TeslaMate.Terrain terminating
teslamate-1  | ** (DBConnection.ConnectionError) tcp recv: closed (the connection was closed by the pool, possibly due to a timeout or because the pool has been terminated)
teslamate-1  |     (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:1047: Ecto.Adapters.SQL.raise_sql_call_error/1
teslamate-1  |     (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:945: Ecto.Adapters.SQL.execute/6
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:229: Ecto.Repo.Queryable.execute/4
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
teslamate-1  |     (teslamate 1.30.1) lib/teslamate/log.ex:177: TeslaMate.Log.get_positions_without_elevation/2
teslamate-1  |     (teslamate 1.30.1) lib/teslamate/terrain.ex:77: TeslaMate.Terrain.handle_event/4
teslamate-1  |     (stdlib 5.2.3) gen_statem.erl:1397: :gen_statem.loop_state_callback/11
teslamate-1  |     (stdlib 5.2.3) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
teslamate-1  | 2024-08-07 11:31:13.241 [error] Postgrex.Protocol (#PID<0.2452.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.2709.0> timed out because it queued and checked out the connection for longer than 60000ms
teslamate-1  | 
teslamate-1  | #PID<0.2709.0> was at location:
teslamate-1  | 
teslamate-1  |     :prim_inet.recv0/3
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:3188: Postgrex.Protocol.msg_recv/4
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:2213: Postgrex.Protocol.recv_bind/3
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:2105: Postgrex.Protocol.bind_execute/4
teslamate-1  |     (db_connection 2.6.0) lib/db_connection/holder.ex:354: DBConnection.Holder.holder_apply/4
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:1512: DBConnection.run_execute/5
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:1607: DBConnection.run/6
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:743: DBConnection.parsed_prepare_execute/5
teslamate-1  | 
teslamate-1  | 2024-08-07 11:31:13.250 [error] GenStateMachine :"1" terminating
teslamate-1  | ** (DBConnection.ConnectionError) tcp recv: closed (the connection was closed by the pool, possibly due to a timeout or because the pool has been terminated)
teslamate-1  |     (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:1047: Ecto.Adapters.SQL.raise_sql_call_error/1
teslamate-1  |     (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:945: Ecto.Adapters.SQL.execute/6
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:229: Ecto.Repo.Queryable.execute/4
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:151: Ecto.Repo.Queryable.one/3
teslamate-1  |     (teslamate 1.30.1) lib/teslamate/vehicles/vehicle.ex:1171: TeslaMate.Vehicles.Vehicle.restore_last_known_values/2
teslamate-1  |     (teslamate 1.30.1) lib/teslamate/vehicles/vehicle.ex:339: TeslaMate.Vehicles.Vehicle.handle_event/4
teslamate-1  |     (stdlib 5.2.3) gen_statem.erl:1397: :gen_statem.loop_state_callback/11
teslamate-1  | 2024-08-07 11:32:13.257 [error] Postgrex.Protocol (#PID<0.2451.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.2730.0> timed out because it queued and checked out the connection for longer than 60000ms
teslamate-1  | 
teslamate-1  | #PID<0.2730.0> was at location:
teslamate-1  | 
teslamate-1  |     :prim_inet.recv0/3
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:3188: Postgrex.Protocol.msg_recv/4
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:2213: Postgrex.Protocol.recv_bind/3
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:2105: Postgrex.Protocol.bind_execute/4
teslamate-1  |     (db_connection 2.6.0) lib/db_connection/holder.ex:354: DBConnection.Holder.holder_apply/4
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:1512: DBConnection.run_execute/5
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:1607: DBConnection.run/6
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:743: DBConnection.parsed_prepare_execute/5
teslamate-1  | 
teslamate-1  | 2024-08-07 11:32:13.771 [error] GenStateMachine TeslaMate.Terrain terminating
teslamate-1  | ** (DBConnection.ConnectionError) tcp recv: closed (the connection was closed by the pool, possibly due to a timeout or because the pool has been terminated)
teslamate-1  |     (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:1047: Ecto.Adapters.SQL.raise_sql_call_error/1
teslamate-1  |     (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:945: Ecto.Adapters.SQL.execute/6
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:229: Ecto.Repo.Queryable.execute/4
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
teslamate-1  |     (teslamate 1.30.1) lib/teslamate/log.ex:177: TeslaMate.Log.get_positions_without_elevation/2
teslamate-1  |     (teslamate 1.30.1) lib/teslamate/terrain.ex:77: TeslaMate.Terrain.handle_event/4
teslamate-1  |     (stdlib 5.2.3) gen_statem.erl:1397: :gen_statem.loop_state_callback/11
teslamate-1  |     (stdlib 5.2.3) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
teslamate-1  | 2024-08-07 11:32:14.248 [error] Postgrex.Protocol (#PID<0.2448.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.2731.0> timed out because it queued and checked out the connection for longer than 60000ms
teslamate-1  | 
teslamate-1  | #PID<0.2731.0> was at location:
teslamate-1  | 
teslamate-1  |     :prim_inet.recv0/3
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:3188: Postgrex.Protocol.msg_recv/4
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:2213: Postgrex.Protocol.recv_bind/3
teslamate-1  |     (postgrex 0.17.3) lib/postgrex/protocol.ex:2105: Postgrex.Protocol.bind_execute/4
teslamate-1  |     (db_connection 2.6.0) lib/db_connection/holder.ex:354: DBConnection.Holder.holder_apply/4
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:1512: DBConnection.run_execute/5
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:1607: DBConnection.run/6
teslamate-1  |     (db_connection 2.6.0) lib/db_connection.ex:743: DBConnection.parsed_prepare_execute/5
teslamate-1  | 
teslamate-1  | 2024-08-07 11:32:14.266 [error] GenStateMachine :"1" terminating
teslamate-1  | ** (DBConnection.ConnectionError) tcp recv: closed (the connection was closed by the pool, possibly due to a timeout or because the pool has been terminated)
teslamate-1  |     (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:1047: Ecto.Adapters.SQL.raise_sql_call_error/1
teslamate-1  |     (ecto_sql 3.10.2) lib/ecto/adapters/sql.ex:945: Ecto.Adapters.SQL.execute/6
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:229: Ecto.Repo.Queryable.execute/4
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
teslamate-1  |     (ecto 3.10.3) lib/ecto/repo/queryable.ex:151: Ecto.Repo.Queryable.one/3
teslamate-1  |     (teslamate 1.30.1) lib/teslamate/vehicles/vehicle.ex:1171: TeslaMate.Vehicles.Vehicle.restore_last_known_values/2
teslamate-1  |     (teslamate 1.30.1) lib/teslamate/vehicles/vehicle.ex:339: TeslaMate.Vehicles.Vehicle.handle_event/4
teslamate-1  |     (stdlib 5.2.3) gen_statem.erl:1397: :gen_statem.loop_state_callback/11
teslamate-1  | 2024-08-07 11:33:14.269 [error] Postgrex.Protocol (#PID<0.2453.0>) disconnected: ** (DBConnection.ConnectionError) client #PID<0.2734.0> timed out because it queued and checked out the connection for longer than 60000ms

**Comment by e-tplus** [comment]
Similar problem here, also on a free Google cloud instance.

**Reply by djpalmis** [reply]
Did you check the log? What does it report?

**Reply by e-tplus** [reply]
"docker compose logs teslamate" prints 3 lines after 30 minutes and does not respond any more. Can I download logs somewhere for external inspection?

**Reply by djpalmis** [reply]
I don't know, sorry. But this is a really bad behaviour.
Actually I can't event transfer files from Google cloud..
